crates/sui-indexer/src/indexer_reader.rs

```diff
@@ -247,6 +257,312 @@ impl IndexerReader {
                 .into_sui_system_state_summary();
         Ok(system_state)
     }
+
+    pub async fn get_owned_object(
```

Comment:
This method should probably be non-async and then wrapped in a call to spawn blocking...or alternatively we could use the new async query method that was recently added: https://github.com/MystenLabs/sui/blob/indexer-v2/crates/telemetry-subscribers/src/lib.rs#L123

Reply:
thinking on this further...the async query function is useful but may provide a false sense of security since it won't be able to help cover cases where we are doing things like module resolution for example

Reply:
I redo a bunch of functions in this file in this way:
1. the complicated query functions (not really composable ones) have `spawn_blocking` in it and named like `xyz_in_blocking_task`
2. leave the composable (e.g. get objects, get transactions) ones as is
3. others are not async